static/skin/taskbar.js

```diff
@@ -1,5 +1,35 @@
+function htmlDecode(input) {
+    var doc = new DOMParser().parseFromString(input, "text/html");
+    return doc.documentElement.textContent;
+}
 
 (function ($) {
+    $( "#kiwixsearchbox" ).autocomplete({
+
+        source: `/suggest?content=${window.location.pathname.split('/')[1] || ''}`,
```

Comment:
`window.location.pathname.split('/')[1]` assumes an empty root location under which the library is served (search for [`m_root` in src](https://github.com/kiwix/kiwix-lib/search?q=m_root) and for `{{root}}` in `static`).

Reply:
Also taking the root location into account the suggestion `source` URL should start with `/${root}/suggest?...`

Reply:
@veloman-yunkan I get the problem but what solution do you have in mind is it possible to somehow get the value of {{root}} in JS

Reply:
One hack is to look for the `<link>` elements in the `<head>` section and deduce the root location. A somewhat cleaner solution is to inject a dedicated link pointing to the the root explicitly (something like: `<link type="root" href="rooturl">`.

Reply:
@veloman-yunkan is this fine ?
```
source: `${root}/suggest?content=${window.location.pathname.split(`${root}/`)[1] || ''}`
```

Reply:
Assuming that `root` is not empty, `window.location.pathname.split(`${root}/`)[1]` won't produce a correct value for the book filename (for `http://localhost/myzimlibrary/zimfile/A/welcome.html` it will evaluate to `zimfile/A/welcome.html`)

Reply:
Test your changes by starting `kiwix-serve` with a non-empty `-r` option, for example:

```
kiwix-serve -p 8080 -r myzimlibrary ...
```

Reply:
Sorry for that😅
I have 2 ways in mind rn to get it working correctly

1) keep the script tag we deleted with just this code
```
<script>const content={{#urlencoded}}{{{content}}}{{/urlencoded}}</script>
```
And use it wherever needed.
But, would this destroy the whole reason of moving content to `taskbar.js`?
As we don't want js code in headhtml.

2) change the already applied method to work properly

```
source:  `${root}/suggest?content=${(root ? window.location.pathname.split(`${root}/`)[1].split('/')[0] : window.location.pathname.split(`/`)[1])  || ''}`
```
In this case is it confirm that content won't have any `/` and will be just simple SIM file names?
I will also add encodeURI to this code if you suggest that 
Please let me know what I should do.

Reply:
I prefer the second option. Yet, doesn't

```
window.location.pathname.split(`${root}/`)[1].split('/')[0]
```
always work (including the case of an empty `root`)?

> In this case is it confirm that content won't have any `/` and will be just simple zim file names?

Looks like so:

https://github.com/kiwix/kiwix-lib/blob/533541cf4592db041c7759a6a8384a054f4a13c0/src/book.cpp#L178-L195

(complemented by the fact that `/` cannot be used in filenames under Windows).

Reply:
ok then will use the second method.....I misunderstood your earlier comment 😅

> Assuming that `root` is not empty, `window.location.pathname.split(`${root}/`)[1]` won't produce a correct value for the book filename (for `http://localhost/myzimlibrary/zimfile/A/welcome.html` it will evaluate to `zimfile/A/welcome.html`)

and used ternary operator for root.
will update the PR.
thanks